File: solutions/javascript/JumpGame/JumpGame.js
/**
 * @param {number[]} nums
 * @return {boolean}
 */
var canJump = function(nums) {
    let res = 0;
    for (let i = 0; i < nums.length; i++) {
        if (i > res) return false;
        res = Math.max(res, i + nums[i]);
    }
    return true;
};

File: solutions/java/JumpGame/JumpGame.java
class Solution {
    public boolean canJump(int[] nums) {
        int res = 0;//记录能够达到的最远index
        for (int i = 0; i < nums.length; i++) {
            if (i > res) {
                return false;
            }
            res = Math.max(res, i + nums[i]);
        }
        return true;
    }
}